### PR TITLE
feat(Integrations): Add Amplitude metrics when requesting an integration install

### PR DESCRIPTION
**Pull request by ceorourke**
This PR will add tracking to when a user without integration install permissions wants to request that an integration be installed, as well as when someone with permission clicks through from the resulting email to install said integration.

### CODE REVIEW

File: src/sentry/static/sentry/app/views/organizationIntegrations/docIntegrationDetailedView.tsx [hunk 1]
```diff
@@ -67,6 +67,12 @@ class SentryAppDetailedView extends AbstractIntegrationDetailedView<
   }
 
   trackClick = () => {
+    if (window.location.href.includes('referrer=request_email')) {
+      this.trackIntegrationEvent({
+        eventKey: 'integrations.install_from_request',
```
Comment (scefali) [inline src/sentry/static/sentry/app/views/organizationIntegrations/docIntegrationDetailedView.tsx:72]: @ceorourke Why do we need a new event for this? We are tracking the `referrer` for all events so we can easily just look at the installation event when `custom_referrer= request_email`.

File: src/sentry/api/endpoints/organization_integration_request.py [hunk 2]
```diff
@@ -101,7 +101,8 @@ def post(self, request, organization):
             html_template="sentry/emails/requests/organization-integration.html",
             type="organization.integration.request",
             context={
-                "integration_link": get_url(organization, provider_type, provider_slug),
+                "integration_link": get_url(organization, provider_type, provider_slug)
+                + "/?referrer=request_email",
```
Comment (scefali) [inline src/sentry/api/endpoints/organization_integration_request.py:105]: @ceorourke This should be part of the `get_url` function

File: src/sentry/api/endpoints/organization_integration_request.py [hunk 3]
```diff
@@ -31,6 +31,7 @@ def get_url(organization, provider_type, provider_slug):
                     "sentry_app": "sentry-apps",
                 }.get(provider_type),
                 provider_slug,
+                "/?referrer=request_email",
```
Comment (scefali) [inline src/sentry/api/endpoints/organization_integration_request.py:34]: @ceorourke I wouldn't expect you need to have the extra `/` here if you're doing a join on `/`.

Comment (ceorourke) [reply src/sentry/api/endpoints/organization_integration_request.py:34]: oof you're right

**Review by scefali — APPROVED** [review]

**Pull request merged**